lib/rucio/client/uploadclient.py

```diff
@@ -726,23 +733,28 @@ def _register_file(
         file_name = file['did_name']
         file_did = {'scope': file_scope, 'name': file_name}
         replica_for_api = self._convert_file_for_api(file)
-        try:
-            # if the remote checksum is different, this DID must not be used
-            meta = self.client.get_metadata(file_scope, file_name)
-            logger(logging.INFO, 'File DID already exists')
-            logger(logging.DEBUG, 'local checksum: %s, remote checksum: %s' % (file['adler32'], meta['adler32']))
 
-            if str(meta['adler32']).lstrip('0') != str(file['adler32']).lstrip('0'):
-                logger(logging.ERROR,
+        if not new_file:
+            try:
+                # if the remote checksum is different, this DID must not be used
+                meta = self.client.get_metadata(file_scope, file_name)
+                logger(logging.INFO, 'File DID already exists')
+                logger(logging.DEBUG, 'local checksum: %s, remote checksum: %s' % (file['adler32'], meta['adler32']))
+
+                if str(meta['adler32']).lstrip('0') != str(file['adler32']).lstrip('0'):
+                    logger(logging.ERROR,
                        'Local checksum %s does not match remote checksum %s' % (file['adler32'], meta['adler32']))
-                raise DataIdentifierAlreadyExists
-
-            # add the file to rse if it is not registered yet
-            replicastate = list(self.client.list_replicas([file_did], all_states=True))
-            if rse not in replicastate[0]['rses']:
-                self.client.add_replicas(rse=rse, files=[replica_for_api])
-                logger(logging.INFO, 'Successfully added replica in Rucio catalogue at %s' % rse)
-        except DataIdentifierNotFound:
+                    raise DataIdentifierAlreadyExists
+
+                # add the file to rse if it is not registered yet
+                replicastate = list(self.client.list_replicas([file_did], all_states=True))
+                if rse not in replicastate[0]['rses']:
+                    self.client.add_replicas(rse=rse, files=[replica_for_api])
+                    logger(logging.INFO, 'Successfully added replica in Rucio catalogue at %s' % rse)
+            except DataIdentifierNotFound:
+                new_file = True
+
+        if new_file:
             logger(logging.DEBUG, 'File DID does not exist')
```

Comment:
1. As a minimal safety mechanism, should there be a warning log for the user instead:

```suggestion
      logger(logging.WARNING, 
           f'--new-file flag used: Skipping validation for {file_scope}:{file_name}. '
           f'If this DID already exists, data corruption will occur.')
```